File: submarinealert/RandomSub.java
import java.util.ArrayDeque;
import java.util.Random;

public class RandomSub implements Submarine {

  int position;
  Random rand = new Random(123);

  ArrayDeque<Integer> commandList = new ArrayDeque<>();

  RandomSub(int position) {
    this.position = position;
  }

  void generateNewCommand() {
    int magnitude = rand.nextInt(10) + 1;
    int dir = rand.nextInt(2) == 0 ? -1 : 1;
    for (int i = 0; i < magnitude; i++) {
      commandList.add(1 * dir);
    }
  }

  public int getMove() {
    if (commandList.size() == 0) {
      generateNewCommand();
    }
    return commandList.poll();
  }

  public void hasBeenProbed(boolean probed) {

  }

}

File: submarinealert/py/clients/random_sub.py
import random
import collections

from clients.submarine_captain_client import SubmarineCaptain

class RandomSub(SubmarineCaptain):

    def __init__(self, position: int):
        super().__init__("Subby McSubFace")
        self.commandList = collections.deque()

    def your_algorithm(self, times_probed):
        return self.getMove()

    def generateNewCommand(self):
        magnitude = random.randint(1, 10)
        dir = random.choice([-1, 1])
        for _ in range(magnitude):
            self.commandList.append(1 * dir)

    def getMove(self) -> int:
        if len(self.commandList) == 0:
            self.generateNewCommand()
        return self.commandList.popleft()
